java/src/main/java/com/scalar/admin/RequestCoordinator.java

```diff
@@ -52,25 +59,35 @@ public JsonObject checkPaused() {
 
   private Map<String, String> run(Function<AdminClient, Optional<String>> function) {
     // Assume that the list of addresses for unpause is the same as the one for pause.
-    List<SRVRecord> records = getApplicationIps(srvServiceUrl);
+    if (addresses == null) {
+      List<SRVRecord> records = getApplicationIps(srvServiceUrl);
+      addresses =
+          records.stream()
```

Comment:
```suggestion
          getApplicationIps(srvServiceUrl);.stream()
```
No need for the `records` variable?

Reply:
Yeah, it could be shorter.